apsi-enrollment-be/src/main/java/edu/pw/apsienrollment/event/api/EventController.java

```diff
@@ -0,0 +1,97 @@
+package edu.pw.apsienrollment.event.api;
+
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import lombok.NonNull;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import lombok.RequiredArgsConstructor;
+
+import edu.pw.apsienrollment.event.EventService;
+import edu.pw.apsienrollment.event.db.Event;
+import edu.pw.apsienrollment.event.specification.EventSpecificationBuilder;
+
+@RestController
+@RequestMapping("events")
+@RequiredArgsConstructor
+public class EventController {
+    private final EventService eventService;
+
+    @GetMapping("")
+    ResponseEntity<Iterable<EventDto>> events(
+            @RequestParam Optional<Integer> page,
+            @RequestParam Optional<Integer> size,
+            @RequestParam Optional<String> search) {
+        if(search.isPresent()) {
+            if(checkPageParams(page, size)) {
+                return this.search(search.get(), page.get(), size.get());
+            }
+            return this.search(search.get());
+        }
+        if(checkPageParams(page, size)) {
+            return findAll(page.get(), size.get());
+        }
+        return findAll();
+    }
+
+    static Boolean checkPageParams(Optional<Integer> page, Optional<Integer> size) {
+        if(page.isPresent() || size.isPresent()) {
+            if(page.isPresent() && size.isPresent()) {
+                if(page.get() < 0) {
```

Comment:
This validation can be replaced with built-in spring validation. You can gather all the query parameters into DTO class, then set conditions of validation using javax annotations such as @Min, @NotNull etc. Default values can be set in the class too. To use DTO object as carrier of query params, simply use it as a method argument and remove @RequestParam annotation.